include/af/image.h

```diff
@@ -677,6 +677,32 @@ AFAPI array ycbcr2rgb(const array& in, const YCCStd standard=AF_YCC_601);
 AFAPI array rgb2ycbcr(const array& in, const YCCStd standard=AF_YCC_601);
 #endif
 
+#if AF_API_VERSION >= 34
+/**
+   C++ Interface for calculating an image moment
+
+   \param[in]  in is the input image
+   \param[moment] is the moment to calculate
+   \return      the value of the moment
+
+   \ingroup image_func_moments
+ */
+template<typename T> T moment(const array& in, const af_moment_type moment);
```

Comment:
@syurkevi There might be a similar comment down somewhere, add a typedef such as `typedef af_moment_type MomentType` under namespace af and use that here.